packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -382,8 +382,9 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain: boolean) {
     };
   }, [scopeRef, contain]);
 
+  // Changed useEffect to useLayoutEffect so this is guaranteed to run before our async synthetic blur
```

Comment:
Code comments should be for future readers, not PR reviewers. Once this is merged, the "Changed useEffect to useLayoutEffect" part won't really make sense anymore.

```suggestion
  // This is a useLayoutEffect so it is guaranteed to run before our async synthetic blur
```